tribe-ext-divi-theme-compatibility.php

```diff
@@ -0,0 +1,272 @@
+ * Version:           1.2.0
+ * Extension Class:   Tribe__Extension__Divi_Theme_Compatibility
+ * Author:            Modern Tribe, Inc.
+ * Author URI:        http://m.tri.be/1971
+ * License:           GPL version 3 or any later version
+ * License URI:       https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ *     This plugin is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     any later version.
+ *
+ *     This plugin is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *     GNU General Public License for more details.
+ */
+
+// Do not load unless Tribe Common is fully loaded and our class does not yet exist.
+if (
+	class_exists( 'Tribe__Extension' )
+	&& ! class_exists( 'Tribe__Extension__Example' )
+) {
+	/**
+	 * Extension main class, class begins loading on init() function.
+	 */
+	class Tribe__Extension__Divi_Theme_Compatibility extends Tribe__Extension {
+		/**
+		 * Setup the Extension's properties.
+		 *
+		 * This always executes even if the required plugins are not present.
+		 */
+		public function construct() {
+			$this->add_required_plugin( 'Tribe__Events__Main' );
+		}
+
+		/**
+		 * Extension initialization and hooks.
+		 */
+		public function init() {
+			// Load plugin textdomain
+			load_plugin_textdomain( 'tribe-ext-divi-theme-compatibility', false, basename( dirname( __FILE__ ) ) . '/languages/' );
+
+			/**
+			 * All extensions require PHP 5.6+, following along with https://theeventscalendar.com/knowledgebase/php-version-requirement-changes/
+			 */
+			$php_required_version = '5.6';
+
+			if ( version_compare( PHP_VERSION, $php_required_version, '<' ) ) {
+				if (
+					is_admin()
+					&& current_user_can( 'activate_plugins' )
+				) {
+					$message = '<p>';
+					$message .= sprintf( __( '%s requires PHP version %s or newer to work. Please contact your website host and inquire about updating PHP.', 'tribe-ext-divi-theme-compatibility' ), $this->get_name(), $php_required_version );
+					$message .= sprintf( ' <a href="%1$s">%1$s</a>', 'https://wordpress.org/about/requirements/' );
+					$message .= '</p>';
+					tribe_notice( $this->get_name(), $message, 'type=error' );
+				}
+
+				return;
+			}
+
+			if ( function_exists( 'et_custom_posts_per_page' ) && ! is_admin() ) {
+				add_filter( 'parse_query', array( $this, 'remove_et_custom_posts_per_page' ), 100 );
+			}
+
+			add_action( 'wp_head', array( $this, 'fix_divi_widget_styles' ) );
+			add_action( 'wp_head', array( $this, 'fix_db_widget_styles' ) );
```

Comment:
Please move the conditionals here so the actions are only added when needed.

Reply:
Yeah, I was thinking about that and stupidly I didn't ask.

Reply:
Not stupid at all.